### DISCUSSION TITLE
Manual/SmartMerge

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#58] [Notes]
# Manual/SmartMerge

https://docs.unity3d.com/Manual/SmartMerge.html

**Comment by vertxxyz** [comment]
If you get an error message referring you to a `mergespecfile.txt` file, it is easier to configure this via Unity's Preferences.
`External Tools/Revision Control Diff/Merge` configures the fallback merger that Smart Merge uses. Once configured Smart Merge will open this tool when a merge couldn't be totally completed. You will only have to resolve the conflicts where Smart Merge failed.

**Comment by joonjoonjoon** [comment]
I was curious if this needed to be kept up to date, but it seems like they haven't upgrade YAML merge since 2017  

```csharp
turbo:~ joon$ /Applications/Unity/Hub/Editor/2017.4.12f1/Unity.app/Contents/Tools/UnityYAMLMerge  
Error: need 'merge' command specified  
Unity YAML Merge Tool for scene or prefab files - version 1.0.1  
  
turbo:~ joon$ /Applications/Unity/Hub/Editor/2019.3.1f1/Unity.app/Contents/Tools/UnityYAMLMerge  
Error: need 'merge' command specified  
Unity YAML Merge Tool for scene or prefab files - version 1.0.1
```

**Comment by miltoncandelero** [comment]
If you are using git, and you set it up correctly (set up the merge tool in the config and the gitattributes for the .unity and .prefab) you still need to USE it. It's not automatic.

Once you get a conflict, you run `git mergetool` and it should find your conflicting scenes and prefabs and fix them for you. Leaving you with 6 versions

- The finally merged one
- The base one (where yours and theirs diverged)
- Yours
- Theirs
- The original file with git merge markers

In case you don't like how unity merged them, you get all the tools to fix the merge.

Maybe this is obvious for most people, but it wasnt for me.
